src/Cache.php

```diff
@@ -25,15 +25,32 @@ class Cache extends BaseCache
     private $versionParser;
     private $symfonyRequire;
     private $symfonyConstraints;
-    private $io;
 
     public function setSymfonyRequire(string $symfonyRequire, array $versions, IOInterface $io = null)
     {
         $this->versionParser = new VersionParser();
         $this->symfonyRequire = $symfonyRequire;
         $this->symfonyConstraints = $this->versionParser->parseConstraints($symfonyRequire);
+
+        foreach ($versions['splits'] as $name => $vers) {
+            foreach ($vers as $i => $v) {
+                $v = $this->versionParser->normalize($v);
+
+                if (!$this->symfonyConstraints->matches(new Constraint('==', $v))) {
+                    if (null !== $io) {
+                        $io->writeError(sprintf('<info>Restricting packages listed in "symfony/symfony" to "%s"</info>', $this->symfonyRequire));
+                        $io = null;
+                    }
+                    unset($vers[$i]);
+                }
+            }
+
+            if (!$vers || $vers === $versions['splits'][$name]) {
+                unset($versions['splits'][$name]);
```

Comment:
we keep only the packages that are affected by the version requirements and have a non-empty intersection with existing versions of it